packages/vite-plugin-cloudflare/src/cloudflare-environment.ts

```diff
@@ -125,6 +134,12 @@ export function createCloudflareEnvironmentOptions(
 	userConfig: vite.UserConfig,
 	environmentName: string
 ): vite.EnvironmentOptions {
+	const workerCompatFlags = workerConfig.compatibility_flags;
+
+	const workerdPopulatesEnv =
+		workerCompatFlags.includes("nodejs_compat") &&
+		workerCompatFlags.includes("nodejs_compat_populate_process_env");
```

Comment:
this should be true also if nodejs_compat is true and compat date >= 2025-04-01

Reply:
ah, ok thanks 🙂 

I was looking for a compat date in the [workerd PR](https://github.com/cloudflare/workerd/pull/3610) but couldn't find one, I assumed this wouldn't be turned on by default automatically then 🤔

Reply:
sorry I was multitasking and looking at the wrong PR! 🤦‍♂️ 😅 

https://github.com/cloudflare/workerd/blob/bd8abeabdf633bb274bf7e0e7f53a9e8656e465e/src/workerd/io/compatibility-date.capnp#L720

thanks again 🙂

Reply:
Updated, please have another look @threepointone 😄